cppcache/src/PdxHelper.cpp

```diff
@@ -47,92 +47,105 @@ PdxHelper::PdxHelper() {}
 
 PdxHelper::~PdxHelper() {}
 
-void PdxHelper::serializePdx(
-    DataOutput& output, const std::shared_ptr<PdxSerializable>& pdxObject) {
-  auto pdxII = std::dynamic_pointer_cast<PdxInstanceImpl>(pdxObject);
-  auto cacheImpl = CacheRegionHelper::getCacheImpl(output.getCache());
-  auto pdxTypeRegistry = cacheImpl->getPdxTypeRegistry();
-  auto& cachePerfStats = cacheImpl->getCachePerfStats();
+std::shared_ptr<PdxInstanceImpl> PdxHelper::getPdxInstance(
```

Comment:
I fear that hiding the `std::dynamic_pointer_cast` and the potential for it go return `nullptr` is going to make for issues later.